Applications/Cxx/gdcmanon.cxx

```diff
@@ -173,28 +156,79 @@ static bool AnonymizeOneFile(gdcm::Anonymizer &anon, const char *filename, const
   gdcm::FileMetaInformation &fmi = file.GetHeader();
   fmi.Clear();
 
-  gdcm::Writer writer;
-  if (strcmp(outfilename, "stdout") == 0) {
-    writer.SetStream(std::cout);
-  } else {
-    writer.SetFileName(outfilename);
-  }
   writer.SetFile( file );
-  if( !writer.Write() )
-    {
-    std::cerr << "Could not Write : " << outfilename << std::endl;
-    if( strcmp(filename,outfilename) != 0 )
-      {
-      gdcm::System::RemoveFile( outfilename );
+
+  return true;
+}
+
+static bool AnonymizeOneFile(gdcm::Anonymizer &anon, const char *filename, const char *outfilename, bool continuemode) {
+    gdcm::Reader reader;
+    reader.SetFileName(filename);
+
+    if (!reader.Read()) {
+      std::cerr << "Could not read : " << filename << std::endl;
+      if (continuemode) {
+        std::cerr << "Skipping from anonymization process (continue mode)." << std::endl;
+        return true;
+      } else {
+        std::cerr << "Check [--continue] option for skipping files." << std::endl;
+        return false;
       }
-    else
-      {
-      std::cerr << "gdcmanon just corrupted: " << filename << " for you (data lost)." << std::endl;
+    }
+
+    gdcm::Writer writer;
+    writer.SetFileName(outfilename);
+    if (!AnonymizeOneFile(anon, reader, writer, filename)) {
+      return false;
+    }
+
+    if (!writer.Write()) {
+      std::cerr << "Could not Write : " << outfilename << std::endl;
+      if (strcmp(filename, outfilename) != 0) {
+        gdcm::System::RemoveFile(outfilename);
+      } else {
+        std::cerr << "gdcmanon just corrupted: " << filename << " for you (data lost)." << std::endl;
       }
+      return false;
+    }
 
-    return false;
+    return true;
+}
+
+static void anonymize_network_files(gdcm::Anonymizer &anon, unsigned short port)
+{
+  io_service ioservice;
+  tcp::endpoint endpoint(tcp::v4(), port);
+  tcp::acceptor acceptor(ioservice, endpoint);
+  std::cout << "Listening for connections on: " << port << std::endl;
+
+  while (true) {
+    tcp::iostream socket_stream;
+    boost::system::error_code connection_error;
+    acceptor.accept(*socket_stream.rdbuf(), connection_error);
+    fixed_istream_buffer seek_buffer(socket_stream, 512);
```

Comment:
How was 512 chosen? This is a magic number, it would be good to give it a name. Does changing this impact performance at all? 512 seems pretty small.

Reply:
This is an arbitrary number larger than the size needed to seek around a dicom header. I did some smoke testing and did not notice a performance hit, but I can investigate in more detail.

Reply:
Here are some numbers:
unbuffered file min 111 max 177 average 130.36 total 26072
512 buffered tcp min 107 max 159 average 120.77 total 24154
1024 buffered tcp min 107 max 153 average 121.64 total 24328
2048 buffered tcp min 106 max 166 average 123.69 total 24738
4096 buffered tcp min 107 max 162 average 121.115 total 24223